### PR TITLE
Removed useless code

### PR DESCRIPTION
**Pull request by marcstern**
Mainly useless memset

### CODE REVIEW

File: apache2/msc_reqbody.c [hunk 1]
```diff
@@ -461,37 +461,30 @@ apr_status_t modsecurity_request_body_to_stream(modsec_rec *msr, const char *buf
...
 
-    memset(msr->stream_input_data, 0, msr->stream_input_length+1);
-
     if(first_pkt)   {
         memcpy(msr->stream_input_data, buffer, msr->stream_input_length);
     } else {
         memcpy(msr->stream_input_data, data, msr->stream_input_length - buflen);
         memcpy(msr->stream_input_data+(msr->stream_input_length - buflen), buffer, buflen);
     }
+    msr->stream_input_data[msr->stream_input_length] = '\0';
```
Comment (airween) [inline apache2/msc_reqbody.c:485]: If I'm right the target string's length can be 2 different value. But here it handled only the first case - will this be good?

File: apache2/re.c [hunk 2]
```diff
@@ -326,14 +326,14 @@ char *update_rule_target_ex(modsec_rec *msr, msre_ruleset *ruleset, msre_rule *r
                     if(value != NULL && targets[i]->param != NULL)  {
                         if((strlen(targets[i]->param) == strlen(value)) &&
                                 strncasecmp(targets[i]->param,value,strlen(targets[i]->param)) == 0) {
-                            memset(targets[i]->name,0,strlen(targets[i]->name));
-                            memset(targets[i]->param,0,strlen(targets[i]->param));
+                            targets[i]->name[0] = '\0';
```
Comment (airween) [inline apache2/re.c:329]: With this evaluation the string will be cleared - was this the intention? Based on old version it seems like the string's length is more than 0.

File: apache2/re_operators.c [hunk 3]
```diff
@@ -1573,12 +1561,11 @@ static const char *gsb_replace_tpath(apr_pool_t *pool, const char *domain, int l
...
     url = apr_palloc(pool, len + 1);
+    if (!url) return NULL;
     data = apr_palloc(pool, len + 1);
+    if (!data) return NULL;
 
-    memset(data, 0, len+1);
-    memset(url, 0, len+1);
-
-    memcpy(url, domain, len);
+    url[len] = 0;
```
Comment (airween) [inline apache2/re_operators.c]: This would be better with `'\0'` (just for better readability).

**Review by airween — CHANGES_REQUESTED** [review]

File: apache2/re.c (hunk 2 above)
Comment (marcstern) [reply apache2/re.c:329]: memset sets all characters to '\0', which is useless, only the first character needs to be set to have a 0-ended string.

File: apache2/msc_reqbody.c (hunk 1 above)
Comment (marcstern) [reply apache2/msc_reqbody.c:485]: First case, we copy `msr->stream_input_length` bytes.
Second case, we copy `msr->stream_input_length - buflen` bytes, then `buflen` bytes.
So, `msr->stream_input_length` bytes in both cases.

Comment (airween) [reply apache2/msc_reqbody.c:485]: Okay, now I reviewed the whole logic and I see. Thanks.

**Review by airween — APPROVED** [review]

**Pull request merged**